library/src/main/java/com/dslplatform/json/processor/Analysis.java

```diff
@@ -1119,13 +1119,13 @@ static TypeMirror unpackType(TypeMirror type, Types types) {
 			TypeMirror sb = unpackType(wt.getSuperBound(), types);
 			return types.getWildcardType(ext, sb);
 		}
-		if (type instanceof DeclaredType && type instanceof TypeElement) {
+		if (type instanceof DeclaredType) {
 			DeclaredType dt = (DeclaredType) type;
 			List<TypeMirror> args = new ArrayList<>(2);
 			for (TypeMirror a : dt.getTypeArguments()) {
 				args.add(unpackType(a, types));
 			}
-			TypeElement te = (TypeElement) dt;
+			TypeElement te = (TypeElement) dt.asElement();

```

Comment:
Not sure what this means in Java, but I would prefer to write something like

    if (type instanceof DeclaredType) {
    	DeclaredType dt = (DeclaredType) type;
    	TypeElement te = type instanceof TypeElement
    			? (TypeElement) type
    			: dt.asElement() instanceof TypeElement
    			? (TypeElement) dt.asElement()
    			: null;
    	if (te != null) {
    		List<TypeMirror> args = new ArrayList<>(2);
    		for (TypeMirror a : dt.getTypeArguments()) {
    			args.add(unpackType(a, types));
    		}
    		return types.getDeclaredType(te, args.toArray(new TypeMirror[0]));
    	}
    }

Reply:
Otherwise looks ok. The only minor improvement would be to fix version in Scala project too (build.sbt)

Reply:
Hi @zapov,

I was on vacation, I see that you already merged, many thanks!
Now may I ask you will you do a new patch release ?

Reply:
I will, probably end of next week